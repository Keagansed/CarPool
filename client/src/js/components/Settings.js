import React, { Component } from 'react';
import { observer } from "mobx-react";
import LoginStore from '../stores/LoginStore';
import { Link } from 'react-router-dom';

import Navbar from './Navbar'
// import 'bootstrap/dist/css/bootstrap.min.css';



@observer class Settings  extends Component {

    handleLogout = () =>
    {
        LoginStore.logOut();
    }

    render(){
        return(
            <div>
                <h1>Settings</h1>
                <Link to={"/"} onClick={this.handleLogout}>Log Out</Link> 
                <Navbar token={ this.props.store.token }/>
            </div>
        );
    }
}

export default Settings;