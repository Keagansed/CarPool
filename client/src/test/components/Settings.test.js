import React from 'react';
import Settings from '../../js/components/Settings';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom'
import configureStore from 'redux-mock-store';

describe('Settings Page Component', () => {
    const mockStore = configureStore();
    let store, container;
    const initialState = {token:"123Foo"};

    beforeEach(() => {
        store = mockStore(initialState);
        container = shallow(<MemoryRouter><Settings store={store} /></MemoryRouter>);
    });

    it('captures snapshot', () => {
        const renderedValue = renderer.create(<MemoryRouter><Settings store={store} /></MemoryRouter>).toJSON();
        expect(renderedValue).toMatchSnapshot();
    });

    it('renders correctly', () => {
        expect(container.length).toEqual(1);
    });
});