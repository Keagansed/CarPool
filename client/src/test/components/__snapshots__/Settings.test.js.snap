// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Page Component captures snapshot 1`] = `
<div
  className="size-100 bg-purple"
>
  <div
    className="fixed-top container-fluid height-50px bg-aqua"
  >
    <div
      className="row font-20px height-100p"
    >
      <button
        className="btnTab  col-6 btn height-100p bg-trans txt-purple fw-bold brad-0 font-20px active"
        onClick={[Function]}
      >
        Profile
      </button>
      <button
        className="btnTab  col-6 btn height-100p bg-trans txt-purple fw-bold brad-0 font-20px "
        onClick={[Function]}
      >
        Alerts
      </button>
    </div>
  </div>
  <div
    className="padtop-50px padbot-50px"
  />
  <div
    className="height-50px fixed-bottom container-fluid pad-10px bg-aqua"
  >
    <div
      className="row"
    >
      <div
        className="col-4 txt-center"
      >
        <a
          href="/Settings"
          onClick={[Function]}
        >
          <i
            className="fa fa-cog txt-purple txt-30px"
          />
        </a>
      </div>
      <div
        className="col-4 txt-center"
      >
        <a
          href="/HomePage"
          onClick={[Function]}
        >
          <i
            className="fa fa-home txt-purple txt-30px"
          />
        </a>
      </div>
      <div
        className="col-4 txt-center"
      >
        <a
          href="/ProfilePage/"
          onClick={[Function]}
        >
          <i
            className="fa fa-user txt-purple txt-30px"
          />
        </a>
      </div>
    </div>
  </div>
</div>
`;
